File: project-template/spring-boot-library-monorepo-gradle/spring-lib-samples/spring-lib-java-sample/src/main/java/acktsap/SomeComponent.java
package acktsap;

import acktsap.spring.lib.Library;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;

@Component
class SomeComponent {
    @Autowired
    private Library library;

    @PostConstruct
    void init() {
        System.out.println("library.someLibraryModuleMethod: " + library.someLibraryModuleMethod());
    }
}

File: project-template/spring-boot-library-monorepo-gradle/spring-lib-samples/spring-lib-kotlin-sample/src/main/kotlin/acktsap/SomeComponent.kt
package acktsap

import org.springframework.stereotype.Component
import javax.annotation.PostConstruct

@Component
open class SomeComponent(
    private val library: Library
) {
    @PostConstruct
    fun init() {
        println("library.someExtension: ${library.someExtension()}")
    }
}
